src/test/java/uk/gov/justice/hmpps/datacompliance/repository/jpa/repository/referral/OffenderDeletionReferralRepositoryTest.java

```diff
@@ -134,24 +152,35 @@ private void assertMatchesExpectedContents(final ReferredOffenderBooking offende
 
     private void assertMatchesExpectedContents(final ReferralResolution resolution) {
         assertThat(resolution.getResolutionDateTime()).isEqualTo(LocalDateTime.of(2021, 2, 3, 4, 5, 6));
-        assertThat(resolution.getResolutionType()).isEqualTo(RETAINED);
-        assertThat(resolution.isType(RETAINED)).isTrue();
+        assertThat(resolution.getResolutionStatus()).isEqualTo(PENDING);
+        assertThat(resolution.isType(PENDING)).isTrue();
 
-        assertMatchesExpectedContents(resolution.getRetentionReasons());
+        assertMatchesExpectedContents(resolution.getRetentionChecks());
     }
 
-    private void assertMatchesExpectedContents(final List<RetentionReason> retentionReasons) {
+    private void assertMatchesExpectedContents(final List<RetentionCheck> retentionChecks) {
+
+        assertThat(retentionChecks).hasSize(3);
 
-        assertThat(retentionReasons).hasSize(2);
-        assertThat(retentionReasons).extracting(RetentionReason::getReasonCode)
-                .containsExactlyInAnyOrder(MANUAL_RETENTION, PATHFINDER_REFERRAL);
+        final var manualRetentionCheck = getRetentionCheck(retentionChecks, MANUAL_RETENTION, RetentionCheckManual.class);
+        assertThat(manualRetentionCheck.getCheckStatus()).isEqualTo(Status.PENDING);
+        assertThat(manualRetentionCheck.getManualRetention().getManualRetentionId()).isEqualTo(1L);
+
+        final var pathfinderReferralCheck = getRetentionCheck(retentionChecks, PATHFINDER_REFERRAL, RetentionCheckPathfinder.class);
+        assertThat(pathfinderReferralCheck.getCheckStatus()).isEqualTo(RETENTION_REQUIRED);
+
+        final var imageDuplicateCheck = getRetentionCheck(retentionChecks, IMAGE_DUPLICATE, RetentionCheckImageDuplicate.class);
+        assertThat(imageDuplicateCheck.getCheckStatus()).isEqualTo(RETENTION_NOT_REQUIRED);
+        assertThat(imageDuplicateCheck.getImageDuplicates()).extracting(ImageDuplicate::getImageDuplicateId).containsExactly(1L);
+    }
 
-        retentionReasons.stream()
-                .filter(reason -> MANUAL_RETENTION.equals(reason.getReasonCode()))
-                .map(RetentionReasonManual.class::cast)
+    private <T extends RetentionCheck> T getRetentionCheck(final List<RetentionCheck> retentionChecks,
+                                                           final String checkType,
+                                                           final Class<T> retentionCheckClass) {
+        return retentionChecks.stream()
+                .filter(check -> checkType.equals(check.getCheckType()))
```

Comment:
As matter of interest could you have just checked the type? Is there a one to one mapping between the class and the checkType?

Reply:
True, would make this a bit simpler, I'll change in next PR